src/pfs/pfs.proto

```diff
@@ -267,57 +267,13 @@ enum Delimiter {
   CSV = 4;
 }
 
-message PutFile {
-  bool append = 1;
+message AddFile {
+  string path = 1;
   string tag = 2;
   oneof source {
-    RawFileSource raw_file_source = 3;
-    TarFileSource tar_file_source = 4;
+    google.protobuf.BytesValue raw_file_source = 4;
```

Comment:
Start at 3 here. It might also make sense to nest the URLFileSource type and just have it be URL, then you could just name the fields raw / data and url.

Reply:
What about the "recursive" url flag?

Reply:
It should be captured in the nested URL type.

Reply:
I don't really understand why you would want to nest anything.  There is no type nesting in Go, so it just obscures the name.  It seems like it's a feature that might work well in Java.

Reply:
It would give us an easier way to scope the type rather than having to suffix with `FileSource`. It would just be a field with type AddFile_URL and name url in the source oneof. Take a look at `PhoneNumber` here: https://developers.google.com/protocol-buffers/docs/gotutorial.